src/Npgsql/NpgsqlBatchCommand.cs

```diff
@@ -152,6 +153,8 @@ internal List<NpgsqlParameter> PositionalParameters
         set => _inputParameters = value;
     }
 
+    internal bool HasParameters => _inputParameters?.Count > 0 || _ownedInputParameters?.Count > 0;
```

Comment:
I understand that it's all for the sake of perf but all these `batchCommand.HasParameters ? batchCommand.PositionalParameters : EmptyParameters` don't look that hot. Maybe we add a property here as a shortcut (something like `CurrentParameters`)?

Reply:
Oh for sure. Removing the parameter allocs was more of a hack job until I finally found all the places, didn't clean it up after, so good suggestion.

Reply:
The other point was that given EmptyParameters is a mutable list on a static it can easily be mutated if its not known. I've opted for CurrentParametersReadOnly now to make that clear.